driver-core/src/main/java/com/datastax/driver/core/Connection.java

```diff
@@ -674,12 +763,17 @@ public void shutdown() {
         int runsWithNoWork = 0;
 
         private Flusher(EventLoop eventLoop) {
-            this.eventLoop = eventLoop;
+            this.eventLoopRef = new WeakReference<EventLoop>(eventLoop);
         }
 
         void start() {
             if (!running.get() && running.compareAndSet(false, true)) {
-                this.eventLoop.execute(this);
+                EventLoop eventLoop = eventLoopRef.get();
+                if(eventLoop == null) {
+                    return;
+                } else {
+                    eventLoop.execute(this);
+                }
```

Comment:
This block can be simplified as:

    if (eventLoop != null)
        eventLoop.execute(this);